File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    
    public int[] topKFrequent(int[] nums, int k) {
PriorityQueue<int[]> pq =
    new PriorityQueue<>((a, b) -> Integer.compare(b[1], a[1]));        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        for(int i:map.keySet()){
            pq.add(new int[]{i,map.get(i)});
        }
        int ans[]=new int[k];
        int i=0;
        while(k-->0){
            
            ans[i]=pq.remove()[0];
            i++;
        }
        return ans;
        
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cs
public class Solution {
class Pair{
    public int n,fr;
    public Pair(int a,int b){
        this.n=a;
        this.fr=b;
    }

}
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int,int> map=new Dictionary<int,int>();
        foreach(int n in nums){
            if(!map.ContainsKey(n)){
                map[n]=0;
            }
            map[n]+=1;
        }
       PriorityQueue<Pair, int> pq = new PriorityQueue<Pair, int>();

        foreach (var e in map)
        {
            pq.Enqueue(new Pair(e.Key, e.Value), -e.Value);
        }

        int[] ans = new int[k];

        for (int i = 0; i < k; i++)
        {
            ans[i] = pq.Dequeue().n;
        }

        return ans;

    }
}